framework/web/AssetManager.php

```diff
@@ -572,10 +594,15 @@ public function getPublishedUrl($path)
      * Generate a CRC32 hash for the directory path. Collisions are higher
      * than MD5 but generates a much smaller hash string.
      * @param string $path string to be hashed.
+     * @param boolean $isFile whether the path is file.
      * @return string hashed string.
      */
-    protected function hash($path)
+    protected function hash($path, $isFile = true)
```

Comment:
Yes, in case of asset files which mostly have extensions like `css`, `jpg`, `js` this makes sense. Going to update this.